kimchi/src/circuits/polynomials/keccak/witness.rs

```diff
@@ -355,16 +355,25 @@ impl Chi {
 /// Values involved in Iota permutation step
 pub struct Iota {
     state_g: Vec<u64>,
+    rc: Vec<u64>,
 }
 
 impl Iota {
-    fn create(state_f: Vec<u64>, round: usize) -> Self {
+    pub fn create(state_f: &[u64], round: usize) -> Self {
         let rc = Keccak::sparse(RC[round]);
-        let mut state_g = state_f.clone();
+        let mut state_g = state_f.to_vec();
```

Comment:
Apparently there is no map in Rust with the index. We could have simplify the code and remove one list iteration.

Reply:
https://github.com/o1-labs/proof-systems/issues/1607